### ISSUE TITLE
LSP: explicit imports don't interact with the references features

### ISSUE THREAD

**Issue by ag-eitilt** [sifive/wake#838]
The title is slightly misleading since I don't actually want the `from x import symbol` to show up in the "Go to references" list or similar, but it would be nice to be able to go to the definition of those symbols without finding their uses in the file below.  More importantly, however, is that since they're not treated as a reference, things like F2 renaming won't affect the list -- renaming something which has been explicitly imported will successfully rename everywhere it's been used, but since the import itself doesn't change the file is promptly broken (can't find the new symbol).

**Comment by terpstra** [comment]
Yeah. The fact that the LSP ignores `import` has number of implications. Unfortunately, the variable resolution algorithm wake uses resolves all references in one pass, so there is no point in the compilation chain for the LSP to look where it would see references that terminate at an import.

So, this is hard to fix.

Rename isn't just broken in that it fails rename the import, but worse, if your import renames the symbol, it will attempt to rename both the old and new names, which totally messes vscode up when those names have differing lengths.

**Comment by ag-eitilt** [comment]
Makes sense.  Shame there's no intermediate hook, but this isn't anything which should come up too frequently and at least the errors should make what happened obvious enough quickly enough.

**Comment by terpstra** [comment]
Just because we can't fix it easily doesn't mean this isn't an issue.
Re-opening so we remember we have this debt and maybe we can fix it down the road.
